rfcs/020 - Dependencies permissions RFC.md

```diff
@@ -51,21 +51,70 @@ This is the basic case when we want to use `new_dep_A` in our project but we do
 
 ![Block at library level](./lib-lvl-v1.png)
 
-The `.dependencies.ln` file could look something like:
-
-Block `@std/cmd` but do not override if any mock exists:
-```ln
-add('https://github.com/org/new_dep_A', some(['cmd']), false);
+The `@std/deps` will have the following API:
+
+```ts
+type Dependency {
+  url: string, // Empty string or ignore it for global blocks at application level
+  block: Array<string>,
+  fullBlock: Array<string>,  // fullBlock, deepBlock, extensiveBlock, exhaustiveBlock?
+}
+
+fn add(url: string) = new Dependency {
+  url: url,
+  block: [],
+  fullBlock: [],
+};
+
+fn add() = new Dependency {
+  url: '',
+  block: [],
+  fullBlock: [],
+};
```

Comment:
This is missing the `add(dep: Dependency)` function that would act on the dependency object and actually do the work. Eg, if they want to list the things they're blocking first and then the library at the end.

Reply:
This is an outdated version, after Colton's comment on Friday.